packages/compiler/src/core/checker.ts

```diff
@@ -163,19 +163,32 @@ import {
 export type CreateTypeProps = Omit<Type, "isFinished" | "entityKind" | keyof TypePrototype>;
 
 export interface Checker {
+  /** @internal */
   typePrototype: TypePrototype;
+  /**
+   * Using this API involve working with the TypeSpec Ast and may change at any time.
+   * See https://typespec.io/docs/handbook/breaking-change-policy/
+   */
   getTypeForNode(node: Node): Type;
```

Comment:
What do people think of the Ast types? What I am thinking is that they all could be under 
```
@typespec/compiler/ast
```

instead to mark the separation. This could include as well some type merging that expose otherwise internal properties.

Reply:
I think this is what we discussed, and it makes sense